mlflow/utils/conda_utils.py

```diff
@@ -0,0 +1,82 @@
+import hashlib
+import json
+import logging
+import os
+from mlflow.utils import process
+
+from mlflow.exceptions import ExecutionException
+# Environment variable indicating a path to a conda installation. MLflow will default to running
+# "conda" if unset
+MLFLOW_CONDA_HOME = "MLFLOW_CONDA_HOME"
+
+_logger = logging.getLogger(__name__)
+
+
+def _activate_conda_env_command(conda_env_name):
+    activate_path = _get_conda_bin_executable("activate")
+    return "source %s %s" % (activate_path, conda_env_name)
```

Comment:
You should use `conda activate` instead of `source activate`.
Lately there has been a lot of issues about it : https://github.com/mlflow/mlflow/issues/1543, https://github.com/mlflow/mlflow/issues/965